test/test_ad9364_p.py

```diff
@@ -101,45 +189,133 @@ def test_ad9364_loopback(test_dma_loopback, iio_uri, classname, channel):
+                tx_hardwaregain_chan0=-20,
+                sample_rate=4000000,
+            ),
+            40,
+        ),
+        (params["one_cw_tone_manual"], 27),
+        (params["one_cw_tone_slow_attack"], 40),
+        (params["change_attenuation_20dB_slow_attack"], 44),
+        (params["change_attenuation_0dB_slow_attack"], 27),
+        (params["change_sampling_rate_60MSPS_slow_attack"], 49),
+        (params["change_sampling_rate_15MSPS_slow_attack"], 51),
+        (params["change_attenuation_10dB_manual"], 40),
+        (params["change_attenuation_5dB_manual"], 32),
+        (params["change_rf_gain_5dB_manual"], 28),
     ],
 )
-@pytest.mark.parametrize("sfdr_min", [40])
 def test_ad9364_sfdr(test_sfdr, iio_uri, classname, channel, param_set, sfdr_min):
     test_sfdr(iio_uri, classname, channel, param_set, sfdr_min)
 
 
+########################################
+@pytest.mark.iio_hardware(hardware, True)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize("channel", [0])
+@pytest.mark.parametrize(
+    "param_set, dds_scale, min_rssi, max_rssi",
+    [
+        (params["one_cw_tone_manual"], 0.12, 24, 27),
+        (params["one_cw_tone_manual"], 0.25, 18, 21),
+        (params["one_cw_tone_manual"], 0.25, 17.5, 21.5),
+        (params["one_cw_tone_manual"], 0.06, 30, 32.5),
+        (params["change_rf_gain_5dB_manual"], 0.25, 21, 24),
+        (params["change_attenuation_10dB_manual"], 0.25, 27, 30),
+        (params["change_attenuation_5dB_manual"], 0.25, 22.5, 25.5),
+    ],
+)
+def test_ad9364_dds_gain_check_vary_power(
+    test_gain_check,
+    iio_uri,
+    classname,
+    channel,
+    param_set,
+    dds_scale,
+    min_rssi,
+    max_rssi,
+):
+    test_gain_check(
+        iio_uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
+    )
+
+
 #########################################
 @pytest.mark.iio_hardware(hardware, True)
 @pytest.mark.parametrize("classname", [(classname)])
 @pytest.mark.parametrize("channel", [0])
-@pytest.mark.parametrize("frequency, scale", [(1000000, 1)])
 @pytest.mark.parametrize(
-    "param_set",
+    "param_set, dds_scale, min_rssi, max_rssi",
     [
-        dict(
-            tx_lo=1000000000,
-            rx_lo=1000000000,
-            gain_control_mode_chan0="slow_attack",
-            tx_hardwaregain_chan0=-30,
-            sample_rate=4000000,
-        )
+        (params["one_cw_tone_slow_attack"], 0.06, 42.5, 45.5),
+        (params["change_attenuation_20dB_slow_attack"], 0.06, 53.5, 55.5),
+        (params["change_attenuation_0dB_slow_attack"], 0.06, 33.5, 35.5),
+    ],
+)
+def test_ad9364_dds_gain_check_agc(
+    test_gain_check,
+    iio_uri,
+    classname,
+    channel,
+    param_set,
+    dds_scale,
+    min_rssi,
+    max_rssi,
+):
+    test_gain_check(
+        iio_uri, classname, channel, param_set, dds_scale, min_rssi, max_rssi
+    )
+
+
+#########################################
+@pytest.mark.iio_hardware(hardware)
```

Comment:
Need to set the ignore emulation flag `@pytest.mark.iio_hardware(hardware. True)`  https://pytest-libiio.readthedocs.io/en/latest/emulation/#device-library

Reply:
I've added two commits below that caters to this. Can you please check if they are on the right functions/tests?